src/components/AssertionForm/ActionButton/ApproveButton.tsx

```diff
@@ -18,10 +41,30 @@ function useApproveButton(props: ActionButtonProps) {
     args: [oracleAddress, bondBigInt],
   });
 
-  const { write } = useContractWrite(config);
+  const { data, write } = useContractWrite(config);
+
+  const { isLoading } = useWaitForTransaction({
+    hash: data?.hash,
+  });
+
+  useEffect(() => {
+    if (data?.hash) {
```

Comment:
this is all we need to do to handle notifications now. calling `useContractWrite` multiple times generates multiple different hashes, and this setup is able to handle multiple concurrent notifications.